app/src/main/java/com/coveros/coverosmobileapp/oauth/OAuthLoginActivity.java

```diff
@@ -0,0 +1,153 @@
+import android.webkit.CookieManager;
+import android.webkit.ValueCallback;
+import android.webkit.WebView;
+
+import com.android.volley.RequestQueue;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.Volley;
+import com.coveros.coverosmobileapp.R;
+import com.coveros.coverosmobileapp.website.CustomWebViewClient;
+
+/**
+ * Displays a WebView that presents user with log-in page.
+ * @author Maria Kim
+ */
+
+@SuppressWarnings("squid:MaximumInheritanceDepth")
+public class OAuthLoginActivity extends AppCompatActivity {
+
+    private static final String AUTH_ENDPOINT = "https://www3.dev.secureci.com/oauth/authorize";
+    private static final String TOKEN_ENDPOINT = "https://www3.dev.secureci.com/oauth/token";
+    private static final String CLIENT_ID = "ZIgsZCSNWZ0R869u10Y7ZNFSpn4y2S";
+    private static final String CLIENT_SECRET = "w1ANtApmollpctwlYxmXFlKke9HiIK";
+    private static final String REDIRECT_URI = "com.coveros.coverosmobileapp://oauthresponse";
+    private static final String GRANT_TYPE = "authorization_code";
+
+    private AccessTokenRequest accessTokenRequest;
+    private AlertDialog errorResponse;
+
+    WebViewAuthCallback webViewAuthCallback = new WebViewAuthCallback();
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.oauth);
+
+        String responseType = "code";
+        AuthUrl authUrl = new AuthUrl(AUTH_ENDPOINT, CLIENT_ID, REDIRECT_URI, responseType);
+
+        WebView login = (WebView) findViewById(R.id.login);
+        Log.d("Auth url", authUrl.toString());
+        login.loadUrl(authUrl.toString());
+
+        setWebViewClient(login, webViewAuthCallback);
+    }
+
+    /**
+     * Sets the WebViewClient that watches the WebView for the redirect URI that contains the authorization code.
+     * @param webView    WebView for which the WebViewClient is set
+     * @param authCallback    callback to wait for the parsing of the authorization code from the redirect uri
+     */
+    private void setWebViewClient(WebView webView, final AuthCallback authCallback) {
+        webView.setWebViewClient(new CustomWebViewClient() {
+            @SuppressWarnings("deprecation")
+            @Override
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                if (url.startsWith(REDIRECT_URI)) {
+                    Uri uri = Uri.parse(url);
+                    authCallback.onSuccess(uri.getQueryParameter("code"));
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        });
+    }
+
+    /**
+     * Creates an AlertDialog to display the response (success or error) from the REST request.
+     * @param context    context on which to display the AlertDialog
+     * @return    AlertDialog with these data
+     */
+    AlertDialog createRequestResponse(Context context) {
```

Comment:
The name is also a bit misleading since I think this is creating an Alert dialog and not a request or response